Before linux-2.6.22, utimensat does not exist in the kernel.
>=glibc-2.6 exposes it, but would always fail if the kernel
does not has an implementation.

Just compiling utimensat does not mean it is usable. Thus
runtime test is necessary.

Index: cmake-3.4.3/Source/kwsys/CMakeLists.txt
===================================================================
--- cmake-3.4.3.orig/Source/kwsys/CMakeLists.txt
+++ cmake-3.4.3/Source/kwsys/CMakeLists.txt
@@ -509,7 +509,7 @@ IF(KWSYS_USE_SystemTools)
     "Checking whether CXX compiler has environ in stdlib.h" DIRECT)
   KWSYS_PLATFORM_CXX_TEST(KWSYS_CXX_HAS_UTIMES
     "Checking whether CXX compiler has utimes" DIRECT)
-  KWSYS_PLATFORM_CXX_TEST(KWSYS_CXX_HAS_UTIMENSAT
+  KWSYS_PLATFORM_CXX_TEST_RUN(KWSYS_CXX_HAS_UTIMENSAT
     "Checking whether CXX compiler has utimensat" DIRECT)
   SET_PROPERTY(SOURCE SystemTools.cxx APPEND PROPERTY COMPILE_DEFINITIONS
     KWSYS_CXX_HAS_SETENV=${KWSYS_CXX_HAS_SETENV}
Index: cmake-3.4.3/Source/kwsys/kwsysPlatformTestsCXX.cxx
===================================================================
--- cmake-3.4.3.orig/Source/kwsys/kwsysPlatformTestsCXX.cxx
+++ cmake-3.4.3/Source/kwsys/kwsysPlatformTestsCXX.cxx
@@ -272,7 +272,7 @@ int main()
 int main()
 {
   struct timespec times[2] = {{0,UTIME_OMIT},{0,UTIME_NOW}};
-  return utimensat(AT_FDCWD, "/example", times, AT_SYMLINK_NOFOLLOW);
+  return utimensat(AT_FDCWD, ".", times, AT_SYMLINK_NOFOLLOW);
 }
 #endif
 
